### PR TITLE
Cyborgs can announce their laws

### PR DESCRIPTION
**Pull request by chromiumboy**
## About the PR

This PR lets cyborgs announce their laws with just the push of a button. Each law can be announced individually, either locally, or to a radio channel to which they can intrinsically transmit

## Why / Balance

This was a requested feature from the Discord (ref https://discord.com/channels/310555209753690112/1140372438501306388)

## Technical details

Updates SiliconLawBuiState so it can send whether the cyborg can speak and the radio channels it can intrinsically transmit to. The LawDisplay add the necessary transmit buttons to each law

## Media

[A demonstration of the system](https://youtu.be/Oam8wurIU0k)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Cyborgs can now state individual laws with the push of a button

### CODE REVIEW

File: Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs [hunk 1]
```diff
@@ -1,21 +1,81 @@
...
+        string lawDescription = Loc.GetString(law.LawString);
+
+        LawNumberLabel.Text = lawIdentifier;
+        LawLabel.SetMessage(lawDescription);
+
+        if (!canVerbalize) // If you can't talk, you can't state your laws...
+            return;
+
+        Button localButton = new Button
+        {
```
Comment (metalgearsloth) [inline Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs:37]:
```suggestion
        var localButton = new Button
        {
```

File: Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs [hunk 2]
```diff
@@ -1,21 +1,81 @@
...
+
+        LawAnnouncementButtons.AddChild(localButton);
+
+        foreach (string radioChannel in radioChannels)
+        {
+            if (!_prototypeManager.TryIndex<RadioChannelPrototype>(radioChannel, out var radioChannelProto))
+                continue;
+
+            Button radioChannelButton = new Button
+            {
```
Comment (metalgearsloth) [inline Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs:58]:
```suggestion
            var radioChannelButton = new Button
            {
```

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs [hunk 3]
```diff
@@ -93,7 +96,13 @@ private void OnToggleLawsScreen(EntityUid uid, SiliconLawBoundComponent componen
...
     private void OnBoundUIOpened(EntityUid uid, SiliconLawBoundComponent component, BoundUIOpenedEvent args)
     {
-        var state = new SiliconLawBuiState(GetLaws(uid));
+        _entityManager.TryGetComponent<IntrinsicRadioTransmitterComponent>(uid, out var intrinsicRadio);
+        _entityManager.TryGetComponent<SpeechComponent>(uid, out var speech);
+
+        bool canSpeak = speech?.Enabled == true;
+        HashSet<string> radioChannels = intrinsicRadio != null ? intrinsicRadio.Channels : new HashSet<string>();
+
+        var state = new SiliconLawBuiState(GetLaws(uid), canSpeak, radioChannels);
```
Comment (metalgearsloth) [inline Content.Server/Silicons/Laws/SiliconLawSystem.cs:105]: Speech component is predicted on the client so CanVerbalize is unnecessary to network.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs:58]: There's also a few other C# style guide corrections needing elsewhere.

Comment (chromiumboy) [reply Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs:58]: Are these issues related to what's covered in https://docs.spacestation14.io/en/getting-started/conventions?

If not, could you please link me to the appropriate style guide? Cheers

**Comment by deltanedas** [comment]
`git restore upstream/master RobustToolbox` assuming your upstream remote is called upstream

**Comment by chromiumboy** [comment]
> `git restore upstream/master RobustToolbox` assuming your upstream remote is called upstream

Thanks, but I just can't get it to work :\ I've updated the toolbox, but it refuses to commit now

**Comment by chromiumboy** [comment]
Couldn't fix the issue with RobustToolbox, so I've opened a new PR. See https://github.com/space-wizards/space-station-14/pull/19474

**Pull request closed without merging**